tests/os_snapshot_test/os_snapshot_windows_test.go

```diff
@@ -46,6 +48,10 @@ func TestShadowCopy(t *testing.T) {
 	}
 
 	sources := clitestutil.ListSnapshotsAndExpectSuccess(t, e)
+
+	require.NotEmpty(t, sources)
+	require.NotEmpty(t, sources[0].Snapshots)
```

Comment:
good check

The existing code is a bit confusing though... would we get at-least one snapshot if we do not have the permission ?

Reply:
Nothing to do here but I find a couple more issues with this test, for example getting any other error besides os.ErrPermission would be suppressed at first and fail later ... not a deal breaker, but I'd prefer to fail fast.

Reply:
I probably should've added a comment to clarify what this is doing. An error is guaranteed here since the all-zero GUID is not a valid shadow copy ID. This is purely an "are you an admin?" check designed to match [what go-vss is doing][isAdmin] to know which version of the test to execute.

[isAdmin]: https://github.com/mxk/go-vss/blob/9b80ac5672d6a81d6a03347c18f2bc01a506c6e9/vss.go#L237-L260

Also, yes, a Kopia snapshot is created in both scenarios. The test would've simply panicked previously if that didn't happen.

Reply:
> I probably should've added a comment to clarify what this is doing. An error is guaranteed here since the all-zero GUID is not a valid shadow copy ID. This is purely an "are you an admin?" check designed to match [what go-vss is doing](https://github.com/mxk/go-vss/blob/9b80ac5672d6a81d6a03347c18f2bc01a506c6e9/vss.go#L237-L260) to know which version of the test to execute.

It makes sense to add the comment for clarity, it makes it easier to understand the test.

> a Kopia snapshot is created in both scenarios. The test would've simply panicked previously if that didn't happen.

Nevertheless, it makes sense to check explicitly than letting the test panic. It makes it easier to track down test failures.